VisualShader crash when trying to add a value to enum of IntParameter

### Tested versions

- Reproducible in: Godot_v4.7-beta3, Godot_v4.7-beta2, Godot_v4.7-beta1
- Not reproducible in: Godot_v4.6.3-stable, Godot_v4.5.1-stable

All these are win64 official builds. Works as expected in previous stable versions.

### System information

Windows 11, Vulkan 1.4.329 - Forward+ - Using Device #0: NVIDIA - NVIDIA GeForce RTX 3080

### Issue description

Editor crashes when pressing `Add Element` of an Enum using the `IntParameter` node in the VisualShader.

Crash log from my actual project which seems to indicate why:
```
--res://assets/tile_shader.tres--
    2 | render_mode blend_mix, depth_draw_opaque, depth_test_default, cull_back, diffuse_lambert, specular_schlick_ggx;
    3 | 
E   4-> instance uniform int NeighborFlags : hint_enum() = 0;
    5 | uniform uint LOD = 0;
    6 | uniform sampler2D AlbedoFlat : source_color, filter_nearest_mipmap_anisotropic;
SHADER ERROR: Expected a string constant.
   at: (null) (res://assets/tile_shader.tres:4)
ERROR: Shader compilation failed.
   at: set_code (servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp:182)
ERROR: Attempted to free invalid ID: 2264989658251325
   at: _free_internal (servers/rendering/rendering_device.cpp:7724)
ERROR: Attempted to free invalid ID: 2264981068316732
   at: _free_internal (servers/rendering/rendering_device.cpp:7724)
ERROR: Attempted to free invalid ID: 2264985363284027
   at: _free_internal (servers/rendering/rendering_device.cpp:7724)

================================================================
CrashHandlerException: Program crashed with signal 11
Engine version: Godot Engine v4.7.beta3.official (86082170822e19ebb619c7f53c8fe54a6f8d3d2a)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
Load address: 7ff6a8980000

[1] 7ff7eca42321 (main+40c2321) - no debug info in PE/COFF executable
[2] 7ffdd039f932 (ntdll.dll+11f932) - no debug info in PE/COFF executable
[3] 7ffdd03e47ae (ntdll.dll+1647ae) - no debug info in PE/COFF executable
[4] 7ffdd02a21c6 (ntdll.dll+221c6) - no debug info in PE/COFF executable
[5] 7ffdd03e40ed (ntdll.dll+1640ed) - no debug info in PE/COFF executable
[6] 7ff7ea39dd57 (main+1a1dd57) - no debug info in PE/COFF executable
[7] 7ff7ed989e7d (main+5009e7d) - no debug info in PE/COFF executable
[8] 7ff7e9ba3757 (main+1223757) - no debug info in PE/COFF executable
[9] 7ff7e9bff95e (main+127f95e) - no debug info in PE/COFF executable
[10] 7ff7e9c075ad (main+12875ad) - no debug info in PE/COFF executable
[11] 7ff7ec77c231 (main+3dfc231) - no debug info in PE/COFF executable
[12] 7ff7ec77c438 (main+3dfc438) - no debug info in PE/COFF executable
[13] 7ff7eae0a122 (main+248a122) - no debug info in PE/COFF executable
[14] 7ff7e89dd43c (main+5d43c) - no debug info in PE/COFF executable
[15] 7ff7e8982ef0 (main+2ef0) - no debug info in PE/COFF executable
[16] 7ff7e89d2501 (main+52501) - no debug info in PE/COFF executable
[17] 7ff7e89d253d (main+5253d) - no debug info in PE/COFF executable
[18] 7ff7e89810c8 (main+10c8) - no debug info in PE/COFF executable
[19] 7ff7e89813f5 (main+13f5) - no debug info in PE/COFF executable
[20] 7ffdcf5be956 (kernel32.dll+2e956) - no debug info in PE/COFF executable
[21] 7ffdd0307c1b (ntdll.dll+87c1b) - no debug info in PE/COFF executable
[22] -1 (main+-7ff7e8980001) - no debug info in PE/COFF executable
-- END OF C++ BACKTRACE --
================================================================
Godot Editor stopped with exit code: 3221225477
```

### Steps to reproduce

- Open MRP
- Press the `Add Element` button under `PackedStringArray`

<img width="631" height="609" alt="Image" src="https://github.com/user-attachments/assets/eb1f7b81-f38a-4f91-9aeb-4b859f49b291" />

### Minimal reproduction project (MRP)

[visualshader-enum-crash.zip](https://github.com/user-attachments/files/28183818/visualshader-enum-crash.zip)

--- a/editor/inspector/editor_properties_array_dict.cpp
+++ b/editor/inspector/editor_properties_array_dict.cpp
@@ -254,6 +254,21 @@ void EditorPropertyArray::initialize_array(Variant &p_array) {
 	}
 }
 
+void EditorPropertyArray::_update_slots_size() {
+	float name_size = 0;
+	for (Slot &slot : slots) {
+		if (name_size == 0) {
+			int max_index = (page_index + 1) * page_length - 1;
+			const String ms = String("M").repeat(itos(max_index).length());
+
+			Ref<Font> font = theme_cache.font;
+			int font_size = theme_cache.font_size;
+			name_size = slots[0].reorder_button->get_minimum_size().x + theme_cache.horizontal_separation + font->get_string_size(ms, HORIZONTAL_ALIGNMENT_LEFT, -1, font_size).x;
+		}
+		slot.prop->set_name_fixed_size(name_size);
+	}
+}
+
 void EditorPropertyArray::_property_changed(const String &p_property, Variant p_value, const String &p_name, bool p_changing) {
 	if (!p_property.begins_with("indices")) {
 		return;
@@ -494,7 +509,6 @@ void EditorPropertyArray::update_property() {
 		paginator->update(page_index, max_page);
 		paginator->set_visible(max_page > 0);
 
-		float name_size = 0;
 		for (Slot &slot : slots) {
 			bool slot_visible = &slot != &reorder_slot && slot.index < size;
 			slot.container->set_visible(slot_visible);
@@ -553,16 +567,10 @@ void EditorPropertyArray::update_property() {
 				callable_mp(slot.prop, &EditorProperty::grab_focus).call_deferred(0);
 				changing_type_index = EditorPropertyArrayObject::NOT_CHANGING_TYPE;
 			}
-			if (name_size == 0) {
-				int max_index = (page_index + 1) * page_length - 1;
-				const String ms = String("M").repeat(itos(max_index).length());
-
-				Ref<Font> font = theme_cache.font;
-				int font_size = theme_cache.font_size;
-				name_size = slots[0].reorder_button->get_minimum_size().x + theme_cache.horizontal_separation + font->get_string_size(ms, HORIZONTAL_ALIGNMENT_LEFT, -1, font_size).x;
-			}
-			slot.prop->set_name_fixed_size(name_size);
 			slot.prop->update_property();
+		}
+		if (is_inside_tree()) {
+			_update_slots_size();
 		}
 
 		updating = false;
@@ -801,6 +809,7 @@ void EditorPropertyArray::_notification(int p_what) {
 					slot.remove_button->set_button_icon(get_editor_theme_icon(SNAME("Remove")));
 				}
 			}
+			_update_slots_size();
 		} break;
 		case NOTIFICATION_DRAG_BEGIN: {
 			if (is_visible_in_tree()) {

--- a/editor/inspector/editor_properties_array_dict.h
+++ b/editor/inspector/editor_properties_array_dict.h
@@ -133,6 +133,7 @@ class EditorPropertyArray : public EditorProperty {
 	int reorder_to_index = -1;
 	float reorder_mouse_y_delta = 0.0f;
 	void initialize_array(Variant &p_array);
+	void _update_slots_size();
 
 	void _page_changed(int p_page);
 

